injected/src/features/web-compat.js

```diff
@@ -202,36 +202,71 @@ export class WebCompat extends ContentFeature {
         if (window.Notification) {
             return;
         }
+
         // Expose the API
+        // window.Notification polyfill is intentionally incompatible with DOM lib types
+        const NotificationConstructor = function Notification(title) {
+            throw new TypeError("Failed to construct 'Notification': Illegal constructor");
+        };
+
         this.defineProperty(window, 'Notification', {
-            value: () => {
-                // noop
-            },
+            value: NotificationConstructor,
             writable: true,
             configurable: true,
             enumerable: false,
         });
 
-        this.defineProperty(window.Notification, 'requestPermission', {
-            value: () => {
-                return Promise.resolve('denied');
-            },
-            writable: true,
+        const toStringFunc = function toString() {
+            return 'function Notification() { [native code] }';
+        };
+
+        this.defineProperty(/** @type {any} */ (window.Notification), 'toString', {
+            value: toStringFunc,
+            writable: false,
             configurable: true,
-            enumerable: true,
+            enumerable: false,
+        });
+
+        this.defineProperty(/** @type {any} */ (window.Notification), 'prototype', {
```

Comment:
Is this one needed? `NotificationConstructor` should get a prototype of Object by default iirc.

I ask because if we do need it, this will also get a prototype by default which I don't think would be expected. You'd have `Notification.prototype.prototype `accessible. (This can be avoided with `Object.create(null)` instead of `{}`).

But overall, I don't think this whole `defineProperty` is needed but validate that.